Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/AbstractDocSubmissionAuditTransforms.java

```diff
@@ -0,0 +1,146 @@
+            participantObject.setParticipantObjectID(pid);
+        }
+        return participantObject;
+    }
+
+    private ParticipantObjectIdentificationType createSubmissionSetParticipantObjectIdentification(String submissionId) {
+
+        ParticipantObjectIdentificationType participantObject = createParticipantObject(
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_TYPE_CODE_ROLE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_CODE,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_CODE_SYSTEM,
+            DocSubmissionAuditTransformsConstants.PARTICIPANT_SUBMISSION_SET_OBJ_ID_TYPE_DISPLAY_NAME);
+
+        if (NullChecker.isNotNullish(submissionId)) {
+            participantObject.setParticipantObjectID(submissionId);
+        }
+        return participantObject;
+    }
+
+    private ParticipantObjectIdentificationType createParticipantObject(short objTypeCodeSys, short objTypeCodeRole,
+        String objIdTypeCode, String objIdTypeCodeSys, String objIdTypeDisplayName) {
+
+        return createParticipantObjectIdentification(objTypeCodeSys, objTypeCodeRole,
+            objIdTypeCode, objIdTypeCodeSys, objIdTypeDisplayName);
+    }
+
+    private String getIdValue(ProvideAndRegisterDocumentSetRequestType request, String idType) {
+        String idValue = null;
+        RegistryObjectType registryObj = extractRegistryObject(
+            request.getSubmitObjectsRequest().getRegistryObjectList());
+        if (registryObj != null && NullChecker.isNotNullish(registryObj.getExternalIdentifier())) {
+            idValue = getIdFromExternalIdentifiers(registryObj.getExternalIdentifier(), idType);
+        }
+        return idValue;
+    }
+
+    private String getIdFromExternalIdentifiers(List<ExternalIdentifierType> externalIdentifiers, String type) {
+        String id = null;
+        for (ExternalIdentifierType identifier : externalIdentifiers) {
+            if (identifier.getName() != null
+                && NullChecker.isNotNullish(identifier.getName().getLocalizedString())
+                && identifier.getName().getLocalizedString().get(0) != null
+                && identifier.getName().getLocalizedString().get(0).getValue().equals(type)) {
+                id = identifier.getValue();
+                break;
+            }
+        }
+        return id;
+    }
+
+    private RegistryObjectType extractRegistryObject(RegistryObjectListType registryList) {
+        RegistryObjectType registryObj = null;
+        if (registryList != null && NullChecker.isNotNullish(registryList.getIdentifiable())) {
+            List<JAXBElement<? extends IdentifiableType>> identifiers = registryList.getIdentifiable();
+            for (JAXBElement<? extends IdentifiableType> object : identifiers) {
```

Comment:
Can you remove "identifiers" and use "registryList.getIdentifiable()" as part of the enhanced for loop?